src/layout.js

```diff
@@ -208,9 +208,12 @@ export function getLengthUnits(length) {
 /**
  * Returns the numeric value of a CSS length value.
  * @param {!LengthDef|string} length
- * @return {number}
+ * @return {number|undefined}
  */
 export function getLengthNumeral(length) {
+  if (!length) {
```

Comment:
> Hmm we are using isNan and Number.isNaN in our code already

https://github.com/ampproject/amphtml/issues/3970

> could the length value be 100% here?

It could, but we definitely not return `100` (which we would by only testing `isNaN`). A better test may be:

```js
length == parseFloat(length) ? length : undefined;
```

Notice that's using `==` for the implicit typecast.